src/pybind/mgr/stats/fs/perf_stats.py

```diff
@@ -452,7 +463,10 @@ def extract_query_filters(self, cmd):
         return FilterSpec(mds_ranks, client_id, client_ip)
 
     def get_perf_data(self, cmd):
-        filter_spec = self.extract_query_filters(cmd)
+        try:
+            filter_spec = self.extract_query_filters(cmd)
+        except ValueError:
+            return -errno.EINVAL, "", "invalid mds_rank or client_id or client_ip"
```

Comment:
could you use the exception string that the functions throw? that was this returns a targeted error string and not a generic one.

Reply:
I kept "invalid mds_rank or client_id or client_ip" because the other exceptions sometime doesn't make much sense to the user.
Eg: Error EINVAL: invalid literal for int() with base 10: ''

Hence I thought to have simple, generic and user friendly message.

Reply:
> Eg: Error EINVAL: invalid literal for int() with base 10: ''

The regexp should fail matching the input and throw `ValueError`, isn't it?

Reply:
Done. Updated regex and removed generic message.